service/src/java/org/apache/hive/service/cli/operation/Operation.java

```diff
@@ -102,7 +109,7 @@ protected Operation(HiveSession parentSession,
         MetricsConstant.COMPLETED_OPERATION_PREFIX, state);
     queryState = new QueryState.Builder()
                      .withConfOverlay(confOverlay)
-                     .withGenerateNewQueryId(true)
+                     .withGenerateNewQueryId(!embedded)
```

Comment:
An hplsql script contains both declarative (for example select) and imperative statements (for example a for loop). The script is executed by an HplSqlOperation but this will start SQLOperations when it sees hive ql stuffs. So this flag indicates that one operation was started by an other operation.